### ISSUE TITLE
Can't get the demo to run

### ISSUE THREAD

**Issue by stabenfeldt** [dwyl/elixir-auth-microsoft#51]
Hi guys,

Thanks for writing this software! Just what I need to convince my colleagues at this MS-heavy place. 
I tried to clone the repo and run the demo, but that did't work out of the box:

` mix deps && mix phx.server``
![image](https://github.com/dwyl/elixir-auth-microsoft/assets/181272/77122ea3-f1f8-457c-8e47-678e572c5fec)

**Comment by nelsonic** [comment]
Hi @stabenfeldt 👋 
You definitely need the environment variables to be set for it to run. 👌
Have you defined them as per the docs? 💭
https://github.com/dwyl/elixir-auth-microsoft?tab=readme-ov-file#2-create-an-app-registration-in-azure-active-directory-

**Comment by stabenfeldt** [comment]
Hi @nelsonic,

Yes, I forgot to mention that I have them set:

```
➜  elixir-auth-microsoft git:(main) ✗ cd demo
direnv: loading ~/Work/elixir-auth-microsoft/demo/.envrc
direnv: export +MICROSOFT_CLIENT_ID +MICROSOFT_CLIENT_SECRET +MICROSOFT_SCOPES_LIST
```

**Comment by nelsonic** [comment]
Ok. Please run: 

  mix setup

That :http error indicates that Erlang/OTP is not fully available which is bizarre if the mix command works at all ...

Please share your error messages as you see them. 👌

**Comment by stabenfeldt** [comment]
`mix setup` ran without problems
`mix phx.server` stopped at:
```
src/ssl_verify_fun_cert_helpers.erl:56:1: Warning: function select_extension/2 is unused
%   56| select_extension(Id, Extensions) ->
%     | ^

src/ssl_verify_fun_cert_helpers.erl:64:1: Warning: function extract_dns_names_from_alt_names/2 is unused
%   64| extract_dns_names_from_alt_names([ExtValue | Rest], Acc) ->
%     | ^

could not compile dependency :ssl_verify_fun, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile ssl_verify_fun --force", update it with "mix deps.update ssl_verify_fun" or clean it with "mix deps.clean ssl_verify_fun"

```

**Comment by stabenfeldt** [comment]
`mix deps.compile ssl_verify_fun --force` 
gave the same error, but `mix deps.update ssl_verify_fun` got all the deps installed.

**Comment by stabenfeldt** [comment]
`mix phx.server` ended here:

```
[info] Running AppWeb.Endpoint with cowboy 2.9.0 at 127.0.0.1:4000 (http)
[debug] Downloading esbuild from https://registry.npmjs.org/esbuild-darwin-arm64/-/esbuild-darwin-arm64-0.14.29.tgz
[info] Access AppWeb.Endpoint at http://localhost:4000
[error] Task #PID<0.2777.0> started from AppWeb.Endpoint terminating
** (UndefinedFunctionError) function :http_util.timestamp/0 is undefined (module :http_util is not available)
    (inets 9.1) :http_util.timestamp()
    (inets 9.1) httpc.erl:750: :httpc.handle_request/9
    (esbuild 0.5.0) lib/esbuild.ex:297: Esbuild.fetch_body!/1
    (esbuild 0.5.0) lib/esbuild.ex:209: Esbuild.install/0
    (esbuild 0.5.0) lib/esbuild.ex:189: Esbuild.install_and_run/2
    (phoenix 1.6.14) lib/phoenix/endpoint/watcher.ex:19: Phoenix.Endpoint.Watcher.watch/2
    (elixir 1.15.7) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
Function: &Phoenix.Endpoint.Watcher.watch/2
    Args: ["esbuild", {Esbuild, :install_and_run, [:default, ["--sourcemap=inline", "--watch"]]}]

```

**Comment by stabenfeldt** [comment]
```
 elixir --version
Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit] [dtrace]

Elixir 1.15.7 (compiled with Erlang/OTP 24)

```

**Comment by stabenfeldt** [comment]
```
➜  demo git:(main) ✗ asdf --version
v0.14.0
➜  demo git:(main) ✗ asdf list
elixir
  1.14.5-otp-26
  1.14.5
  1.15.7-otp-26
 *1.15.7
  1.16
erlang
  No versions installed
```

**Comment by LuchoTurtle** [comment]
Hey @stabenfeldt !

Super sorry you're having trouble running the demo! It seems that `mix.lock` has outdated dependencies. 
I'm pushing a PR asap to the `main` branch so you have it working!

**Comment by LuchoTurtle** [comment]
Opened #52 to fix this issue. It works on that branch. Waiting on @nelsonic to approve and merge the changes.

If you can't wait for it to get merged, you can checkout the branch and test the demo yourself :)

Tell us if it's working =D

**Comment by stabenfeldt** [comment]
Thanks for helping out, @LuchoTurtle and @nelsonic! 

The application starts without any problems now, but it throws an exception when I press the "Sign in" button.
![image](https://github.com/dwyl/elixir-auth-microsoft/assets/181272/a196db4d-d4cf-4d4b-9158-66021fc6f1df)

**Comment by LuchoTurtle** [comment]
🤔 Interesting, that doesn't seem to be happening to me. 
The only related issue I can find is https://github.com/danhper/plug-navigation-history/issues/6.

What is your app name? We usually give our apps ":app" for simplicity's sake but there's a small chance it may be related to the issue you've raised. The `plug` dependency seems to be using the browser storage to save cookies and it has a hard set cap.

**Comment by nelsonic** [comment]
Issue automatically closed by merging PR #52 … ✅
Reopening for further feedback from @stabenfeldt 💭

**Comment by stabenfeldt** [comment]
I didn't get the demo to work, but when I followed your guide I were able to add SSO to my existing app.
🎉
